pkg/composableschemadsl/compiler/importer.go

```diff
@@ -0,0 +1,54 @@
+package compiler
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/authzed/spicedb/pkg/composableschemadsl/input"
+	"github.com/authzed/spicedb/pkg/genutil/mapz"
+)
+
+type ImportContext struct {
+	pathSegments []string
+	sourceFolder string
+	names        *mapz.Set[string]
+}
+
+const SchemaFileSuffix = ".zed"
+
+func ImportFile(importContext *ImportContext) (*CompiledSchema, error) {
+	relativeFilepath := constructFilePath(importContext.pathSegments)
+	filePath := path.Join(importContext.sourceFolder, relativeFilepath)
```

Comment:
what happens if the imports are outside of the root context folder? is that supported, and if not, what kind of opinions/guidelines does it impose on the structure of a composable schema?

Reply:
No; the eventual goal will be to support this using git-based imports

Reply:
The biggest effect that I see is that you won't be able to do a relative import *within* the package that walks up a directory. this means that you couldn't have:

```
app1/app1.zed
app2/app2.zed
util.zed
```

and have both app files reference the util file. I have a feeling this will be a tripping point; it'll either have to be well-documented or something we change.